nemo_aligner/models/nlp/gpt/megatron_gpt_critic.py

```diff
@@ -48,11 +47,12 @@ class MegatronGPTCriticModel(MegatronGPTRewardModel, CriticModelInterface):
     def __init__(self, cfg: DictConfig, trainer: Trainer):
         super().__init__(cfg, trainer=trainer)
 
-        # filled by the examples script
-        self.rm_state_dict = None
+        # must be populated by the examples script
+        self.rm_state_dict_cpu = None
 
-        # for the critic states on cpu
-        self.cpu_state_dict = None
+        self.critic_state_dict_cpu = copy_model_states_to_cpu(
```

Comment:
it should take around half a second on a GPU with HBM. Yeah, the only time this comes up is if the user doesn't have a critic checkpoint already.

what can be done though is maybe do this on the fly, like if onload_checkpoint is not called then we can load it again. but that is a bit messy since we'll need to load on the fly if `_load_critic` or `finish_training` is called for whatever reason

Reply:
Ok half a second sounds fine to me.